impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -120,6 +126,14 @@ public int compare(ValueHolder<?> t, ValueHolder<?> u) {
   private final Class<V> valueType;
   private final Copier<K> keyCopier;
   private final Copier<V> valueCopier;
+  
+  private final SizeOfEngine sizeOfEngine;
+  private final long faultSizeInBytes;
```

Comment:
Do we really need all these fields here? Can't they be encapsulated somewhere else?